src/server/services/procedures/smithsonian/smithsonian.js

```diff
@@ -0,0 +1,113 @@
+ApiConsumer.setRequiredApiKey(Smithsonian, DataDotGovKey);
+
+function listify(item) {
+    return item === undefined ? [] : item instanceof Array ? item : [item];
+}
+
+Smithsonian._raw_search = function(term, count, skip, filter) {
+    return this._requestData({path:'search', queryString:`api_key=${this.apiKey.value}&q=${term}&start=${skip}&rows=${count}`}).then(res => {
+        const items = [];
+        for (const item of res.response.rows) {
+            if (!filter(item)) continue; // unfortunately they don't have built-in support for our filtering needs
+
+            const notes = [];
+            if ('notes' in item.content.freetext) {
+                for (const note of listify(item.content.freetext.notes)) {
+                    notes.push(note.content);
+                }
+            }
+
+            const physicalDescriptions = [];
+            if ('physicalDescription' in item.content.freetext) {
+                for (const desc of listify(item.content.freetext.physicalDescription)) {
+                    physicalDescriptions.push([desc.label, desc.content]);
+                }
+            }
+
+            const sources = [];
+            if ('dataSource' in item.content.freetext) {
+                for (const src of listify(item.content.freetext.dataSource)) {
+                    sources.push([src.label, src.content]);
+                }
+            }
+
+            const id = item.id;
+            const title = item.title;
+            const types = listify(item.content.indexedStructured.object_type);
+            const authors = listify(item.content.indexedStructured.name);
+            const topics = listify(item.content.indexedStructured.topic);
+
+            items.push({ id, title, types, authors, topics, notes, physicalDescriptions, sources });
+        }
+        return items;
+    });
+};
+
+/**
+ * Search and return up to count matching items
+ * 
+ * @param {String} term Term to search for
+ * @param {BoundedNumber<0>=} count Maximum number of items to return
+ * @param {BoundedNumber<0>=} skip Number of items to skip from beginning
+ * @returns {Array} Up to count matches, each being [id, title, types[], authors[], topics[], notes[], physicalDescriptions[], sources[]]
+ */
+Smithsonian.search = function(term, count = 100, skip = 0) {
+    return this._raw_search(term, count, skip, () => true);
+};
+
+/**
+ * Search and return up to count matching items (only ones with images)
+ * 
+ * @param {String} term Term to search for
+ * @param {BoundedNumber<0>=} count Maximum number of items to return
+ * @param {BoundedNumber<0>=} skip Number of items to skip from beginning
+ * @returns {Array} Up to count matches, each being [id, title, types[], authors[], topics[], notes[], physicalDescriptions[], sources[]]
+ */
+Smithsonian.searchForImages = function(term, count = 100, skip = 0) {
+    return this._raw_search(term, count, skip, item => {
```

Comment:
It would probably be good to update this to actually fetch `count` results after filtering. One way to do this would be to use `async`/`await` and just perform raw_search in a loop until you have either run out of results or found `count` items.